subsys/bluetooth/host/att.c

```diff
@@ -3251,6 +3343,8 @@ static struct bt_att_chan *att_chan_new(struct bt_att *att, atomic_val_t flags)
 	(void)memset(chan, 0, sizeof(*chan));
 	chan->chan.chan.ops = &ops;
 	k_fifo_init(&chan->tx_queue);
+	memset(&chan->rsp_buf, 0, sizeof(chan->rsp_buf));
```

Comment:
`chan->rsp_buf` is a pointer, so it'd be clearer if you just did `chan->rsp_buf = NULL`. That said, setting a reference counted pointer to NULL without unreffing it first looks very suspicious (and the fact that you have a memset here potentially hides such a bug, since the reviewer doesn't necessarily understand that it's a pointer variable you're memsetting to zero).

Reply:
> Since it's the initial initialization, I can't really check `buf->ref` since there might be garbage in there.

Right, sorry, I missed that part. However, you have `(void)memset(chan, 0, sizeof(*chan));` just a few lines earlier, so it seems any additional zeroing is just redundant?

Reply:
that's true, I can also drop the zeroing of the `meta` too